From f86df79e414919fb7a3dc7dbb952a5caaab361f0 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <b29396@freescale.com>
Date: Mon, 13 Jan 2014 16:20:55 +0800
Subject: [PATCH 0215/1594] ENGR00333136 dts: imx6: enable keep power
 capability for corresponding boards

commit 8fbf8e9308009d2827c2e6995ea1152e9a419557 from
git://git.freescale.com/imx/linux-2.6-imx.git

All i.MX6 SabreAuto/SabreSD/EVK has the ability to keep card power
during suspend. So add this capability for them.

Signed-off-by: Dong Aisheng <b29396@freescale.com>
(cherry picked from commit 3f18df4746eb33e934c55de23d6496bb4adad33b)
---
 arch/arm/boot/dts/imx6qdl-sabreauto.dtsi |    1 +
 arch/arm/boot/dts/imx6qdl-sabresd.dtsi   |    3 +++
 arch/arm/boot/dts/imx6sl-evk.dts         |    3 +++
 3 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
index cb7e9ab..08e6646 100644
--- a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
@@ -835,6 +835,7 @@
 	 * Note: reg_sd3_vmmc is also need to be enabled
 	 */
 	/* vmmc-supply = <&reg_sd3_vmmc>; */
+	keep-power-in-suspend;
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
index 4910acf..e22ea41 100644
--- a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
@@ -857,6 +857,7 @@
 	cd-gpios = <&gpio2 2 GPIO_ACTIVE_LOW>;
 	wp-gpios = <&gpio2 3 GPIO_ACTIVE_HIGH>;
 	no-1-8-v;
+	keep-power-in-suspend;
 	status = "okay";
 };
 
@@ -867,6 +868,7 @@
 	cd-gpios = <&gpio2 0 GPIO_ACTIVE_LOW>;
 	wp-gpios = <&gpio2 1 GPIO_ACTIVE_HIGH>;
 	no-1-8-v;
+	keep-power-in-suspend;
 	status = "okay";
 };
 
@@ -884,5 +886,6 @@
 	bus-width = <8>;
 	non-removable;
 	no-1-8-v;
+	keep-power-in-suspend;
 	status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx6sl-evk.dts b/arch/arm/boot/dts/imx6sl-evk.dts
index 9aa8026..5063171 100644
--- a/arch/arm/boot/dts/imx6sl-evk.dts
+++ b/arch/arm/boot/dts/imx6sl-evk.dts
@@ -857,6 +857,7 @@
 	bus-width = <8>;
 	cd-gpios = <&gpio4 7 GPIO_ACTIVE_LOW>;
 	wp-gpios = <&gpio4 6 GPIO_ACTIVE_HIGH>;
+	keep-power-in-suspend;
 	status = "okay";
 };
 
@@ -867,6 +868,7 @@
 	pinctrl-2 = <&pinctrl_usdhc2_200mhz>;
 	cd-gpios = <&gpio5 0 GPIO_ACTIVE_LOW>;
 	wp-gpios = <&gpio4 29 GPIO_ACTIVE_HIGH>;
+	keep-power-in-suspend;
 	status = "okay";
 };
 
@@ -876,5 +878,6 @@
 	pinctrl-1 = <&pinctrl_usdhc3_100mhz>;
 	pinctrl-2 = <&pinctrl_usdhc3_200mhz>;
 	cd-gpios = <&gpio3 22 GPIO_ACTIVE_LOW>;
+	keep-power-in-suspend;
 	status = "okay";
 };
-- 
1.7.5.4

